From fc2d6012e2a9ede1af463e0510254a8706dc7365 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 28 Nov 2016 13:30:44 +0200
Subject: [PATCH 0628/1345] fix: dts: a80x0: fix sdhci for A80x0-DB on CP0

commit  8d85395a1a0c6b8616470b4a7eacb65e12c081f1 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- fix sdhci on CP to use 4 bit mode (CP supports only this mode)
- Add missing SDHCI configuration for AP: 8bit mode

Change-Id: I285e552a0d1c15489a792bcfaecd93f8496a552f
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34130
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 1bba048..86c4689 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -56,6 +56,15 @@
 		reg = <0x0 0x0 0x0 0x80000000>;
 	};
 
+	ap806 {
+		config-space@f0000000 {
+			sdhci@6e0000 {
+				bus-width = <8>;
+				non-removable;
+			};
+		};
+	};
+
 	cp110-master {
 		config-space@f2000000 {
 			i2c@701000 {
@@ -106,6 +115,7 @@
 				usb-phy = <&cpm_usb3h1_phy>;
 			};
 			sdhci@780000 {
+				bus-width = <4>;
 				non-removable;
 			};
 		};
-- 
1.7.9.5

